language/move-prover/boogie-backend/src/bytecode_translator.rs

```diff
@@ -1095,13 +1245,182 @@ impl<'env> FunctionTranslator<'env> {
 
                         // regular path
                         if !processed {
-                            if dest_str.is_empty() {
-                                emitln!(
-                                    writer,
-                                    "call {}({});",
-                                    boogie_function_name(&callee_env, inst),
-                                    args_str
+                            let targeted = self.fun_target.module_env().is_target();
+                            let caller_mid = self.fun_target.module_env().get_id();
+                            let caller_fid = self.fun_target.get_id();
+                            let fun_verified =
+                                !self.fun_target.func_env.is_explicitly_not_verified(
+                                    &ProverOptions::get(self.fun_target.global_env()).verify_scope,
                                 );
+                            if dest_str.is_empty() {
+                                // TODO(tengzhang): a hacky way to generate calls for vector functions, need refactor
+                                if module_env
```

Comment:
Move this test into model_env, and also check the address?

`module_env.is_std_vector()`